pkg/sources/scheduler/persistence/file.go

```diff
@@ -0,0 +1,197 @@
+package persistence
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+	"sync"
+	"time"
+
+	"github.com/dukex/operion/pkg/sources/scheduler/models"
+)
+
+// FilePersistence implements SchedulerPersistence using JSON files.
+type FilePersistence struct {
```

Comment:
I prefer keep the persistence out of here, it's better create a new model called `Source`, with `id` and `configuration` (and so on). The ideia is make it equal the workflows, you can have many sources configured. 

Using this example we can have:

```json
{
   "id": "test-1",
   "source_id": "scheduler",
   "configuration": {
      "polling_interval": "2m",
      "cron_expression": "*/2 * * * *"
   }
}
```

Given it is native, we don't need implement a new persistence layer at each source